### PR TITLE
[Code coverage] Improve the coverage of stream_readable.js

### PR DESCRIPTION
**Pull request by knightburton**
IoT.js-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

### CODE REVIEW

File: test/run_pass/test_stream.js [hunk 1]
```diff
@@ -18,60 +18,152 @@ var Readable = require('stream').Readable;
...
+
+// Push something to the stream while paused.
+readable3.push(str);
+readable3.push(null);
+
+// Resume the stream after 1s,
+// the pushed str and null shuld be processed after this.
+setTimeout(function() {
+  readable3.resume();
+}, 1000);
```
Comment (hs0225) [inline test/run_pass/test_stream.js]: I'm not sure that `pause` and `resume` work well with this test. How about the following?

```js
var paused = false;
var str = 'test';

readable3.on('data', function(data) {
  assert.equal(paused, true);
  assert.equal(data, str);
});

readable3.pause();
readable3.push(str);
readable3.push(null);

setTimeout(function() {
  paused = true;
  readable3.resume();
}, 1000);
```

Comment (knightburton) [inline test/run_pass/test_stream.js]: Yes, you are right, this is much cleaner than mine.

File: test/run_pass/test_stream.js [hunk 2]
```diff
@@ -18,60 +18,152 @@ var Readable = require('stream').Readable;
...
+}, TypeError);
+
+// Read with irregular parameters from an empty stream.
+assert.equal(readable2.read(-2), null);
+assert.equal(readable2.read(0), null);
+
+readable2.push('qwerty');
+assert.equal(readable2.read(6), 'qwerty');
+
+// Try to read less data than we have.
```
Comment (knightburton) [inline test/run_pass/test_stream.js]: I've changed the commit text.

**Comment by knightburton** [comment]
@hs0225 I've updated the PR based on your comments.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**